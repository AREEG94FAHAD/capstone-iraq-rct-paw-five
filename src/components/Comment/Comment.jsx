import React, { useEffect, useState } from 'react'
import { PropTypes } from 'prop-types'
import { faTrash } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { useDispatch, useSelector } from 'react-redux'
import { deleteComment } from '../../redux/comments/actions'

export default function Comment({ username, usercomment, commentId, userId }) {
  const userState = useSelector(state => state.user.isLoggedIn)
  const user = useSelector(state => state.user)
  const [currentuserId, setcurrentuserId] = useState(
    useSelector(state => state.user.isLoggedIn)
  )

  useEffect(() => {
    if (userState === true) {
      setcurrentuserId(user.user.uid)
    } else {
      setcurrentuserId('')
    }
  }, [userState])

  const dispatch = useDispatch()
  const deletePetComment = (e, Id) => {
    dispatch(deleteComment(Id))
  }
  return (
    <div className="p-2">
      <div className="card " style={{ borderRadius: '20px' }}>
        <div className="card-body">
          <p className="card-text">{username}</p>
          <p className="card-text">{usercomment}</p>
          {userId === currentuserId ? (
            <FontAwesomeIcon
              onClick={e => deletePetComment(e, commentId)}
              icon={faTrash}
              className="float-right text-danger"
            />
          ) : (
            ''
          )}
        </div>
      </div>
    </div>
  )
}
Comment.propTypes = {
  username: PropTypes.string,
  usercomment: PropTypes.string,
  commentId: PropTypes.string,
  userId: PropTypes.string,
}

Comment.defaultProps = {
  username: '',
  usercomment: '',
  commentId: '',
  userId: '',
}
