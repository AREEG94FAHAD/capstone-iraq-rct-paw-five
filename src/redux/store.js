import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import logger from 'redux-logger'
import combineReducers from './reducers/AllReducers'

const store = createStore(
  combineReducers,
  composeWithDevTools(applyMiddleware(logger))
)

export default store
