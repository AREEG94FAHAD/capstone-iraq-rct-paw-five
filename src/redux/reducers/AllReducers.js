import { combineReducers } from 'redux'
import petReducer, {
  langReducer,
  myPets,
  deleteMyPet,
  updateMyPet,
} from './petReducer'
import user from './user'

export default combineReducers({
  langReducer,
  pets: petReducer,
  user,
  myPets,
  deleteMyPet,
  updateMyPet,
})
