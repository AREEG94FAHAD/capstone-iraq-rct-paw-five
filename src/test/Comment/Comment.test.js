import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { BrowserRouter as Router } from 'react-router-dom'
import Comment from '../../components/Comment/Comment'
import AllReducers from '../../redux/reducers/AllReducers'

const store = createStore(AllReducers)

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Comment store={store} />
        </Router>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
