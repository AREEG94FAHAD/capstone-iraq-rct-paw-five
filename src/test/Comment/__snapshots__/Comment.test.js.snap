// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="p-2"
>
  <div
    className="card "
    style={
      Object {
        "borderRadius": "20px",
      }
    }
  >
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        
      </p>
      <p
        className="card-text"
      >
        
      </p>
      
    </div>
  </div>
</div>
`;
